Makefile

```diff
@@ -23,31 +24,19 @@ PATCHES += pam_slurm_adopt-avoid-running-outside-of-the-sshd-PA.patch
 PATCHES += pam_slurm_adopt-send_user_msg-don-t-copy-undefined-d.patch
 PATCHES += pam_slurm_adopt-use-uid-to-determine-whether-root-is.patch
 PATCHES += slurm-rpmlintrc
+else
+SPEC_UPDATE := $(shell cp slurm.spec_centos slurm.spec)
+endif
+ifeq ($(SPEC_UPDATE),)
```

Comment:
And then don't need this.  I am surprised it was even needed at all with you using `:=` in your `SPEC_UPDATE` assignments.  I would have thought this forcing the evaluation of `$(SPEC_UPDATE)` would have only been needed if you assigned to `$(SPEC_UPDATE)` with `=`.

Reply:
I was not aware that I could just put the functions in the makefile to be executed.
From what I think I read about the := assignment is that the assignment happens the first time the target variable is referenced, not the actual assignment statement.

Reply:
Yeah, unless they return a value, you don't need to assign them to anything.

> From what I think I read about the := assignment is that the assignment happens the first time the target variable is referenced, not the actual assignment statement.

￼The opposite.  [`=` vs. `:=`](https://stackoverflow.com/questions/448910/what-is-the-difference-between-the-gnu-makefile-variable-assignments-a)